### PR TITLE
support live migration in GPIO sample

### PR DESCRIPTION
**Pull request by tmakatos**

### CODE REVIEW

File: samples/gpio-pci-idio-16.c [hunk 1]
```diff
@@ -67,6 +71,59 @@ static void _sa_handler(UNUSED int signum)
...
 
+static int
+migration_device_state_transition(vfu_ctx_t *vfu_ctx, vfu_migr_state_t state)
+{
+    vfu_log(vfu_ctx, LOG_DEBUG, "migration: transition to state %d", state);
+    return 0;
+}
+
+static __u64
+migration_get_pending_bytes(vfu_ctx_t *vfu_ctx __attribute__((unused)))
```
Comment (jlevon) [inline samples/gpio-pci-idio-16.c:82]: nit, here, elsewhere: use UNUSED?

File: samples/gpio-pci-idio-16.c [hunk 2]
```diff
@@ -67,6 +71,59 @@ static void _sa_handler(UNUSED int signum)
...
+        *size = sizeof(pin);
+    }
+    return 0;
+}
+
+static ssize_t
+migration_read_data(vfu_ctx_t *vfu_ctx __attribute__((unused)), void *buf,
+                    __u64 size,  __u64 offset)
+{
+    assert (offset == 0 && size == sizeof(pin));
```
Comment (jlevon) [inline samples/gpio-pci-idio-16.c:105]: nit, spurious space

Comment (jlevon) [reply samples/gpio-pci-idio-16.c:105]: let's separate out the two asserts too

File: samples/gpio-pci-idio-16.c [hunk 3]
```diff
@@ -67,6 +71,59 @@ static void _sa_handler(UNUSED int signum)
...
+{
+    assert(count == sizeof(pin));
+    return 0;
+}
+
+static ssize_t
+migration_write_data(vfu_ctx_t *vfu_ctx __attribute__((unused)),
+                     void *buf, __u64 size, __u64 offset)
+{
+    assert(offset == 0 && size == sizeof(pin));
```
Comment (jlevon) [inline samples/gpio-pci-idio-16.c:122]: ditto

**Review by jlevon — CHANGES_REQUESTED** [review]
just a couple of nits

File: samples/gpio-pci-idio-16.c (hunk 1 above)
Comment (tmakatos) [reply samples/gpio-pci-idio-16.c:82]: ok

File: samples/gpio-pci-idio-16.c (hunk 2 above)
Comment (tmakatos) [reply samples/gpio-pci-idio-16.c:105]: ok

**Review by jlevon — APPROVED** [review]

**Pull request merged**